matrix_sdk_crypto/src/verification/sas/helpers.rs

```diff
@@ -538,6 +547,8 @@ pub fn content_to_request(
         AnyToDeviceEventContent::KeyVerificationKey(_) => EventType::KeyVerificationKey,
         AnyToDeviceEventContent::KeyVerificationMac(_) => EventType::KeyVerificationMac,
         AnyToDeviceEventContent::KeyVerificationCancel(_) => EventType::KeyVerificationCancel,
+        AnyToDeviceEventContent::KeyVerificationReady(_) => EventType::KeyVerificationReady,
+        AnyToDeviceEventContent::KeyVerificationDone(_) => EventType::KeyVerificationDone,
         _ => unreachable!(),
```

Comment:
This is not unreachable, `AnyToDeviceEventContent` is non-exhaustive, so this *can* be hit.

Looking at this further, it seems there is an implicit contract for `content_to_request` (which is `pub`, by the way) that it will only get any of these enums, which you cannot be assured is the case.

Reply:
As the name suggests this turns event contents, and verification event contents at that, we created into a request. It's a developer error to call this for unsupported event contents, thus this being unreachable.

Reply:
It's not developer error if you do not correctly note that this can happen, it is not developer error if you *panic* upon this silent assumption, at least maybe return an error detailing this, so that working with this function isn't that painstaking.

It is not the responsibility of the developer to figure this out for themselves, this is rust, it is *your* responsibility to a. Make this unable to happen in the first place, or at least b. *document this*

There is *no* documentation for this function, now tell me again it is "developer error" for a user of this *official SDK* to not know about this by themselves.

Reply:
Do note, this is a private method only used by the verification module, it's public inside the verification module but it isn't exposed to the rest of the SDK nor to the SDK consumers.

So I'm afraid I'm going to tell you again, it's a developer, if you want to be more precise, developer of the SDK error to not know about this.

Please tone down your ideological rhetoric.

Reply:
Alright yeah, I apologise for my tone, I also did not realise that this isn't a publicly-exposed function like I said it was.

That changes my take somewhat, you're correct to say then that it is developer error to not fully immerse themselves in this library *as* a developer of this library, I missed that last point.

My comments were more specific to developing the library and interfaces as exposed to users of this library, in this case, I was wrong, and I was also wrong for my tone (I apologise for that), though I still want to come across that - were this behaviour be a publicly exposed interface - I think that it'd be wrong to assume user developers would be versed in how exactly this library handles their input, and what not to give it (without warning from the library).

From my perspective, it breaks the principles of Locality, that you can figure out what something is supposed to do, own, or handle based on its local description and information, I thought that was broken and disregarded here, but for a library's internals, I acknowledge that - while not exactly good - it's okay to relax it a little.

So yes, sorry for my rhetoric, I got wrapped up in an idea.

Reply:
Anyways, let me rethink and make a specific issue for this case (docs and technical reachability)